crates/provider/src/shared/client.rs

```diff
@@ -0,0 +1,71 @@
+use std::error::Error;
+
+use async_trait::async_trait;
+
+use super::types::{FileMap, NameSpaceDef, PodDef, Settings};
+use crate::shared::types::{RunCommandOptions, RunCommandResponse};
+
+#[async_trait]
+#[allow(non_upper_case_globals)]
+pub trait Client {
```

Comment:
```suggestion
pub trait Provider {
```

I think we need to call it `Provider`, since at the moment we have a wording mix between Provider and Client. I think a good way to reason about the naming is that each `provider` needs to `impl` the `Provider` trait and then interact with a `client` to deploy/manage the internal resources (e.g pods/process).
wdyt?

Reply:
I agree - already changed it from @l0r1s 's comment. Thank you